std/algorithm/comparison.d

```diff
@@ -1007,16 +1011,28 @@ private:
     // Treat _matrix as a rectangular array
     ref CostType matrix(size_t row, size_t col) { return _matrix[row * cols + col]; }
 
-    void AllocMatrix(size_t r, size_t c) {
+    void AllocMatrix(size_t r, size_t c) @trusted {
         rows = r;
         cols = c;
         if (_matrix.length < r * c) {
-            delete _matrix;
-            _matrix = new CostType[r * c];
+            import core.stdc.stdlib : realloc;
+            auto m = cast(CostType *)realloc(_matrix.ptr, r * c * _matrix[0].sizeof);
+            if (!m)
+                assert(0);
```

Comment:
Well, onOutOfMemoryError should do exactly that, report an OOM condition and exit the process. If we don't use it because it based on the EH mechanism, then we should either remove that function or change it's implementation.